redhat-appstudio/service-provider-integration-operator
# disable persistent session and reduce Absolute timeout of a session

### What does this PR do?
- Reduces absolute timeout of a cookie to 1 hour. The absolute timeout is the time after which the session expires, even if the user interacts with the server. Counted from the creation of the cookie (session).

- Makes the session cookie "not persisted." This means the cookie should be stored only in memory by the web client, not on disk. When the client is closed, the cookie should be removed.
Note that this is not completely true as MDN web docs describe it.
![image](https://user-images.githubusercontent.com/73115616/221845168-797390a1-1ca2-41e6-b8bc-4b1d2b1e772d.png)

### Screenshot/screencast of this PR

![image](https://user-images.githubusercontent.com/73115616/221845714-70924601-7b5f-45ce-8d1a-abb74a69da8e.png)


### What issues does this PR fix or reference?

Ensures compliance with [OWASP's Session Management Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#additional-client-side-defenses-for-session-management)
https://issues.redhat.com/browse/SVPI-347


### How to test this PR?

1. Deploy the OAuth service with the image bellow.
2. Create any sample binding `kubectl apply -f samples/binding-dockerconfigjson.yaml --as system:serviceaccount:default:default`
3. use /login endpoint with valid k8s_token and check the cookie. It should not have a `max-age` or `expires` attribute. In my case (picture above), it shows with `Session` value, but the actual set-cookie header does not have these attributes.
4. After you close the browser and reopen it the cookie should be gone, but that did not work in my case as per the MDN comment above.

Additionally, after an hour of interaction with OAuth service (at least one per 5 minutes), the session should also be invalid. But this is harder to test.


```
quay.io/pbaran/spis:SVPI-347-session
```
```
quay.io/redhat-appstudio/pull-request-builds:spi-operator-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
